### PR TITLE
feat: add support for migrating totp

### PR DESCRIPTION
**Pull request by Jonbeckas**
This PR adds the support for a totps attribute in LegacyUser, that allows to migrate multiple Totp devices, that have a base32 encoded secret and are based on SHA1

### CODE REVIEW

**Comment by daniel-frak** [comment]
Hi! Sorry it's taking me so long to look at this one. I'll try to take a closer look around this weekend.

In the meantime, could you document how this TOTP migration works in the `README.md`, if any additional configuration is required etc.?

**Comment by Jonbeckas** [comment]
Sorry, I just read your message. Let me now if you need more documentation.

**Comment by daniel-frak** [comment]
...And so the weekend came and went. Sorry for the delay, again.

Could you add the information you provided in the PR description to the README.md: 
- that this plugin supports migration of multiple Totp devices
- that the devices must have a base32 encoded secret and be based on SHA1

It could even be get its own subheader for TOTP migration. I'm not well-versed in TOTP, but I assume both of these are important.

I'll merge this MR regardless, but, if you have some time, it would be cool if you could reformat the code, as there's currently no auto-formatter configured for this project. Some examples:

- [double line breaks in this file](https://github.com/daniel-frak/keycloak-user-migration/pull/197/files#diff-9b9362a10e8f26f60c36914677bf05442a4f664496cc87e6b2f2cee99e610823R3)
- [no line breaks between methods](https://github.com/daniel-frak/keycloak-user-migration/pull/197/files#diff-9f2c078e603bd9433ac9fdbdc7e5a3012ba5b536726cefc4f05d5211c97d186eR101)
- [no line breaks after class declarations](https://github.com/daniel-frak/keycloak-user-migration/pull/197/files#diff-b049a673471ec607ac6605ae793bf5a2763ce4ee8607f2f9e75f7099738a467bR6)
- [private final properties not kept together](https://github.com/daniel-frak/keycloak-user-migration/pull/197/files#diff-9f2c078e603bd9433ac9fdbdc7e5a3012ba5b536726cefc4f05d5211c97d186eR101)

Last, but not least, a suggestion. This suggests that the "devices must have a base32 encoded secret and be based on SHA1" requirement could be removed by making the otpModel configurable via the UI:

https://github.com/daniel-frak/keycloak-user-migration/pull/197/files#diff-9ec471ee9ea5062b096c9f6cfd6213c56bd97761d2f51d18504c04399764892aR99

I understand if you don't feel like doing it, but at least for future reference, am I correct in that assumption?

**Comment by Jonbeckas** [comment]
Thank you.
Based on the Keycloak code it should be possible to use "HmacSha1", "HmacSHA256", "HmacSHA512"  based totp, with a secret either encoded in BASE32 or UTF-8, with any number of return digets and any period.
I don't know if it is better to configure the type of totp algorithm in the config or dynamicly within the dto from the legacy system.
What do you think?

**Comment by daniel-frak** [comment]
Good question!
Now that I think about it, it might be more flexible to allow TOTP configuration from the legacy system, as that way it could be done per-user (should there be a need like that).

Would you like to do it as part of this MR, or should I merge this in?

**Comment by Jonbeckas** [comment]
I added the configurability now. If your fine with it and the documentation you could merge it.

File: README.md
Comment (daniel-frak) [inline README.md:91]: Some of the properties are missing here.

File: README.md
Comment (daniel-frak) [inline README.md:328]: Using `backticks` for the properties would increase readability. E.g.,:

"`name` should be the name of the totp device (...)".

File: src/test/java/com/danielfrak/code/keycloak/providers/rest/remote/LegacyTotpTest.java
Comment (daniel-frak) [inline src/test/java/com/danielfrak/code/keycloak/providers/rest/remote/LegacyTotpTest.java:17]: Double newline

File: README.md
Comment (daniel-frak) [inline README.md:328]: Grammar nitpick: `the utf8 bytes` →`UTF-8 bytes` (removed `the`, uppercased and kebab-cased `UTF-8`).
Ditto about `the utf8 bytes` below.

Comment (daniel-frak) [inline README.md:328]: I'm not sure if I understand it correctly, but wouldn't `UTF-8 plaintext` be easier to understand than `UTF-8 bytes`?

**Review by daniel-frak — CHANGES_REQUESTED** [review]

**Comment by daniel-frak** [comment]
Great! I added a few nitpicks (mostly about documentation) and I see that Sonar found something, as well.

After that's addressed, I'll gladly merge it in :)

**Comment by Jonbeckas** [comment]
I think i fixed everything

**Comment by daniel-frak** [comment]
Great, thanks for the contribution!

**Pull request merged**
